### ISSUE TITLE
Combobox not showing initial value

### ISSUE THREAD

**Issue by aguahombre** [AvaloniaUI/Avalonia#9046]
**Describe the bug**
If the SelectedItem is set before the Items property, the ComboBox does not display the initial selection.
Swap the order and the initial selection displays correctly.

**To Reproduce**
Run the repo, the ComboBox is displayed with an empty selection.

**Expected behavior**
The ComboBox should show the SelectedItem; i.e. "hello", as this was already set when the bound items property was changed.
The order of setting Items and SelectedItem should not matter.

**Screenshots**

**Desktop (please complete the following information):**

- OS: Windows
- Version 0.11-preview1

**Additional context**
https://github.com/aguahombre/ComboTest.git

**Comment by rabbitism** [comment]
imo it should work like this.

**Comment by robloo** [comment]
I also think this is the intended behavior and WPF (along with other XAML frameworks) is the same. If you set the selected item before the collection the control has no way of knowing what to do with it. It needs the collection of values first.

Here is one of the older discussions for this: https://stackoverflow.com/questions/1301119/silverlight-xaml-attribute-definition-order-matters

I think this should probably be documented but closed as an issue.

**Comment by aguahombre** [comment]
It wouldn't be too difficult to observe Items property changes and set the SelectedIndex to match the SelectedItem.
The control must already be doing this when SelectedItem is set.
This would remove all ordering dependencies and make Avalonia better than WPF.
Why copy bad practice from WPF?

**Comment by timunie** [comment]
I think if you find a reliable way to do, you may provide a PR.

**Comment by rabbitism** [comment]
> It wouldn't be too difficult to observe Items property changes and set the SelectedIndex to match the SelectedItem. The control must already be doing this when SelectedItem is set. This would remove all ordering dependencies and make Avalonia better than WPF. Why copy bad practice from WPF?

The real problem is **match**, how to define match?

**Comment by robloo** [comment]
I don't think this logic is actually in ComboBox itself. You will probably need to go into ItemsControl or perhaps somewhere else. This means you will improve this for many controls at once.

> The real problem is match, how to define match?

"Match" is well defined in this case. object.Equals() is called or any overridden implementation of such. For ref types this compares by reference as default, for value type is uses the default equality method which should be implemented. Works exactly the same as the IList<T> methods for IndexOf(), Contains(), etc.

https://github.com/AvaloniaUI/Avalonia/blob/6a3804bfde7867ab0c419a71d504636f1e196b4b/src/Avalonia.Controls/ItemsControl.cs#L243
